chapter_04/p11_random_node.py

```diff
@@ -86,6 +89,43 @@ def get_random_node(self):
                 raise RuntimeError("Should not be possible")
 
 
+def _delete(node, key):
+
+    if node is None:
+        return node
+
+    if key < node.key:
+        node.left = _delete(node.left, key)
+
+    elif key > node.key:
+        node.right = _delete(node.right, key)
+
+    else:
+        if node.left is None:
+            temp, node = node.right, None
+            return temp
+
+        elif node.right is None:
+            temp, node = node.left, None
+            return temp
+
+        temp = minvalnode(node.right)
+        node.key = temp.key
+        node.right = _delete(node.right, temp.key)
+
+    node.size -= 1
+    return node
+
+
+def minvalnode(node):
```

Comment:
I'll try and improve that linter message. In python the convention is to use underscore_case for function names.  So this shoudl be min_val_node